src/pages/add/select-currency/select-currency.ts

```diff
@@ -187,66 +182,64 @@ export class SelectCurrencyPage {
     });
   }
 
-  // Tokens
-  public setWallets(): void {
-    this.wallets = this.navParam.data.keyId
+  public createAndBindTokenWallet(pairedWallet, token) {
+    if (!_.isEmpty(pairedWallet)) {
+      const tokenWalletClient = this.profileProvider.createTokenWallet(
+        pairedWallet,
+        token
+      );
+      this.profileProvider
+        .addAndBindWalletClient(tokenWalletClient)
+        .then(() => {
+          this.endProcess();
+        });
+    } else {
+      this.endProcess();
+    }
+  }
+
+  public showPairedWalletSelector(token) {
+    const eligibleWallets = this.navParam.data.keyId
       ? this.profileProvider.getWalletsFromGroup({
           keyId: this.navParam.data.keyId,
-          network: 'livenet'
+          network: 'livenet',
+          pairFor: token
         })
       : [];
-    this.walletsEth = this.wallets.filter(wallet => wallet.coin == 'eth');
-  }
 
-  public showWalletSelector(token) {
-    const tokenWalletIds = this.wallets
-      .filter(wallet => wallet.coin === token.symbol.toLowerCase())
-      .map(wallet => wallet.id);
-    const availableWallets = this.wallets.filter(
-      wallet =>
-        !tokenWalletIds.includes(`${wallet.id}-${token.address}`) &&
-        wallet.coin === 'eth'
-    );
+    if (eligibleWallets.length == 1) {
+      return this.createAndBindTokenWallet(eligibleWallets[0], token);
+    }
```

Comment:
Remove this `if block`
Need to show wallet selector even if there is only 1 eth wallet available to confirm that the user is creating a token wallet and to confirm that token wallet is being linked to ETH wallet.